doc/sphinx/Pacemaker_Explained/status.rst

```diff
@@ -1,280 +1,347 @@
+:ref:`node_attributes`. In addition to any values set by the administrator or
+resource agents using the ``attrd_updater`` or ``crm_attribute`` tools, the
+cluster stores various state information here.
          
-.. topic:: A set of transient node attributes for node **cl-virt-1**
+.. topic:: Example transient node attributes for a node
 
    .. code-block:: xml
    
       <transient_attributes id="cl-virt-1">
         <instance_attributes id="status-cl-virt-1">
            <nvpair id="status-cl-virt-1-pingd" name="pingd" value="3"/>
-           <nvpair id="status-cl-virt-1-probe_complete" name="probe_complete" value="true"/>
            <nvpair id="status-cl-virt-1-fail-count-pingd:0.monitor_30000" name="fail-count-pingd:0#monitor_30000" value="1"/>
            <nvpair id="status-cl-virt-1-last-failure-pingd:0" name="last-failure-pingd:0" value="1239009742"/>
         </instance_attributes>
       </transient_attributes>
    
-In the above example, we can see that a monitor on the ``pingd:0`` resource has
-failed once, at 09:22:22 UTC 6 April 2009. [#]_.
 
-We also see that the node is connected to three **pingd** peers and that
-all known resources have been checked for on this machine (``probe_complete``).
-         
 .. index::
-   single: Operation History
+   pair: XML element; lrm
+   pair: XML element; lrm_resources
+   pair: node; history
 
-Operation History
-#################
-   
-A node's resource history is held in the ``lrm_resources`` element (a child
-of the ``lrm`` element). The information stored here includes enough
-information for the cluster to stop the resource safely if it is
-removed from the ``configuration`` section. Specifically, the resource's
-``id``, ``class``, ``type`` and ``provider`` are stored.
+Node History
+############
 
-.. topic:: A record of the ``apcstonith`` resource
+Each ``node_state`` element contains an ``lrm`` element with a history of
+certain resource actions performed on the node. The ``lrm`` element contains an
+``lrm_resources`` element.
 
-   .. code-block:: xml
+.. index::
+   pair: XML element; lrm_resource
+   pair: resource; history
 
-      <lrm_resource id="apcstonith" type="fence_apc_snmp" class="stonith"/>
-   
-Additionally, we store history entries for certain actions.
+Resource History
+________________
+
+The ``lrm_resources`` element contains an ``lrm_resource`` element for each
+resource that has had an action performed on the node.
+
+A ``lrm_resource`` entry has attributes allowing the cluster to stop the
+resource safely even if it is removed from the configuration. Specifically, the
+resource's ``id``, ``class``, ``type`` and ``provider`` are recorded.
+
+.. index::
+   pair: XML element; lrm_rsc_op
+   pair: action; history
+
+Action History
+______________
 
-.. table:: **Attributes of an lrm_rsc_op element**
+Each ``lrm_resource`` element contains an ``lrm_rsc_op`` element for each
+recorded action performed for that resource on that node.
```

Comment:
seems like I normally see them only for monitor, last, and last_failure

Reply:
Right, each *recorded* action. I'll clarify